stac_fastapi/api/stac_fastapi/api/models.py

```diff
@@ -37,11 +38,11 @@ def create_request_model(
 
     mixins = mixins or []
 
-    models = [base_model] + extension_models + mixins
+    models = extension_models + mixins + [base_model]
```

Comment:
⚠️ if we don't do this then dataclass will raise an error because optional fields could be placed before non optional fields 

I don't really like this because it makes mandator that mixing or extensions have only Optional parameters